From: Andrew Morton <akpm@linux-foundation.org>
Subject: zram-add-size-class-equals-check-into-recompression-fix
Date: Mon Oct 24 04:30:52 PM PDT 2022

fix it for "zsmalloc: turn zspage order into runtime variable".

Cc: Alexey Romanov <avromanov@sberdevices.ru>
Cc: Minchan Kim <minchan@kernel.org>
Cc: Nitin Gupta <ngupta@vflare.org>
Cc: Sergey Senozhatsky <senozhatsky@chromium.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/zsmalloc.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/mm/zsmalloc.c~zram-add-size-class-equals-check-into-recompression-fix
+++ a/mm/zsmalloc.c
@@ -1223,7 +1223,9 @@ static bool zspage_full(struct size_clas
  */
 unsigned int zs_get_class_size(struct zs_pool *pool, unsigned int size)
 {
-	struct size_class *class = pool->size_class[get_size_class_index(size)];
+	struct size_class *class;
+
+	class = pool->size_class[get_size_class_index(pool, size)];
 
 	return class->size;
 }
_
